packages/teams-js/src/public/dialog.ts

```diff
@@ -16,21 +18,45 @@ import { runtime } from './runtime';
  *
  * @beta
  */
+export interface Result {
+  err: string;
+  result: string | object;
+}
+
 export namespace dialog {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  type PostMessageChannel = (message: any) => void;
+  type DialogSubmitHandler = (result: Result) => void;
   /**
    * Allows an app to open the dialog module.
    *
    * @param dialogInfo - An object containing the parameters of the dialog module
    * @param submitHandler - Handler to call when the task module is completed
+   * @param messageForChildHandler - Handler that triggers if dialog tries to send a message to the app.
+   *
+   * @returns a Handler that is triggerd to send a message to the dialog.
    */
   export function open(
     dialogInfo: DialogInfo,
-    submitHandler?: (err: string, result: string | object) => void,
-  ): IAppWindow {
+    submitHandler?: DialogSubmitHandler,
+    messageForChildHandler?: PostMessageChannel,
+  ): PostMessageChannel {
     ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
 
-    sendMessageToParent('tasks.startTask', [dialogInfo], submitHandler);
-    return new ChildAppWindow();
+    if (messageForChildHandler) {
+      registerHandler('messageForParent', messageForChildHandler);
```

Comment:
I think there's a separate item tracking this, but just in case: we should make sure to also unregister the handler when dealing with the submit call

Reply:
Yeah, I will do it in a separate call :)